gossip/src/gossip_service.rs

```diff
@@ -96,13 +98,21 @@ impl GossipService {
             gossip_validators,
             exit.clone(),
         );
-        let t_responder = streamer::responder(
+        let t_responder = streamer::responder_atomic(
             "Gossip",
-            gossip_socket,
+            gossip_socket.clone(),
             response_receiver,
             socket_addr_space,
             stats_reporter_sender,
         );
+        let t_rebind = gossip_rebind_rx.map(|rebind_rx| {
```

Comment:
Given that binding a socket takes literally microseconds, do we need a thread waiting on a channel to do it? Can we maybe rebind immediately and just swap the thing in place immediately in the admin RPC handling code?